internal/digraph/executor_validator.go

```diff
@@ -0,0 +1,73 @@
+package digraph
+
+import "fmt"
+
+// ExecutorValidator validates executor-specific configurations
+type ExecutorValidator interface {
+	ValidateStep(step *Step) error
+}
+
+// ExecutorValidatorRegistry maps executor types to their validators
+var executorValidatorRegistry = make(map[string]ExecutorValidator)
+
+// RegisterExecutorValidator registers a validator for an executor type
+func RegisterExecutorValidator(executorType string, validator ExecutorValidator) {
+	executorValidatorRegistry[executorType] = validator
+}
+
+// SSHExecutorValidator validates SSH executor configurations
+type SSHExecutorValidator struct{}
+
+func (v *SSHExecutorValidator) ValidateStep(step *Step) error {
+	// SSH executor doesn't support script field
+	if step.Script != "" {
+		return fmt.Errorf(
+			"script field is not supported with SSH executor. " +
+				"Use 'command' field instead. " +
+				"See: https://github.com/dagu-org/dagu/issues/1306",
+		)
+	}
+	return nil
+}
+
+// DockerExecutorValidator validates Docker executor configurations
+type DockerExecutorValidator struct{}
```

Comment:
What do you think about implementing the validator interface directly within each executor?  
This approach would remove the need for a separate struct and the `RegisterExecutorValidator` function.  
We could then use type assertions to check whether an executor implements the validator interface (see: https://go.dev/ref/spec#Type_assertions).

To complete this refactor, it may make sense to relocate the executor registry to the `digraph` package:  
https://github.com/dagu-org/dagu/blob/71c5d0f6366e079d370caf865420199ec806ebda/internal/digraph/executor/executor.go#L54-L72